dev/structured-logging.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
+
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import os
+import sys
+import re
+import glob
+
+from sparktestsupport import SPARK_HOME
+
+def main():
+    nonmigrated_pattern = r'log(?:Info|Warning|Error)\((?:".*"\.format\(.*\)|s".*(?:\$|\+\s*[^\s"]).*"\))'
+
+    # Regex patterns for file paths to exclude from the Structured Logging style check
+    excluded_file_patterns = [
+        '[Tt]est',
+        '/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala',
+        '/sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIService.scala'
+    ]
+
+    nonmigrated_files = {}
+    excluded_files = set()
+
+    scala_files = glob.glob(os.path.join(SPARK_HOME, '**', '*.scala'), recursive=True)
+
+    for file in scala_files:
+        for exclude_pattern in excluded_file_patterns:
+            if re.search(exclude_pattern, file):
+                excluded_files.add(file)
+
+    for file in scala_files:
+        if file not in excluded_files:
+            with open(file, 'r') as f:
+                lines = f.readlines()
+                for line_number, line in enumerate(lines, start=1):
+                    matches = list(re.finditer(nonmigrated_pattern, line))
```

Comment:
The regex doesn't seem working.

```
import re

nonmigrated_pattern = r'log(?:Info|Warning|Error)\((?:".*"\.format\(.*\)|s".*(?:\$|\+\s*[^\s"]).*"\))'
line = "  logInfo(\"Running Spark version \" + SPARK_VERSION)"
matches = list(re.finditer(nonmigrated_pattern, line)) # matches is empty
```

Reply:
this is still not resolved

Reply:
This is still not working...
If the pattern becomes too complicated, we can start with multiple patters.